### PR TITLE
[#2640] feat(docs): Add the doc for gvfs

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

Add gvfs usage documentation. Currently, three usages in `hadoop shell`, `java code` and `spark program` are introduced.

### Why are the changes needed?

Fix: #2640

### CODE REVIEW

File: docs/how-to-use-gvfs.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+| `fs.gvfs.impl`                                        | The Gravitino Virtual File System implementation class. Please configure it as `com.datastrato.gravitino.filesystem.hadoop3.GravitinoVirtualFileSystem`.                                          | (none)        | Yes      | 0.5.0         |
+| `fs.gvfs.impl.disable.cache`                          | Close the Gravitino Virtual File System cache in Hadoop environment. If you need to proxy multi-user operations, please set this value to `true` and create a separate File System for each user. | `false`       | No       | 0.5.0         |
+| `fs.gravitino.server.uri`                             | The Gravitino server uri which gvfs needs to load the fileset meta.                                                                                                                               | (none)        | Yes      | 0.5.0         |
+| `fs.gravitino.client.metalake`                        | The metalake which fileset belongs.                                                                                                                                                               | (none)        | Yes      | 0.5.0         |
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:34]: I'm not sure `fs.gravitino.client.metalake` is a config option name.

File: docs/how-to-use-gvfs.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+## Introduction
+
+Gravitino supports operating the storage mapping to `fileset` through the Hadoop Compatible File System (HCFS) interface,
+which names `Gravitino Virtual File System` (It is called `gvfs` for short).  
+You can access the storage mapping to fileset through the following path:
+```text
+gvfs://fileset/${catalog_name}/${schema_name}/${fileset_name}/sub_dir/
+```
+The following mainly introduces how to use `Gravitino Virtual File System` to operate fileset which mapping to HDFS.
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:16]: HDFS -> HCFS.

**Comment by qqqttt123** [comment]
Could you polish the document according to https://github.com/datastrato/gravitino/pull/2781

File: docs/how-to-use-gvfs.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    <name>fs.gravitino.client.metalake</name>
+    <value>test_metalake</value>
+  </property>
+```
+
+## Usage
+
+Please get the `Gravitino Virtual File System runtime jar` firstly, you can get it in two ways:
+1. [Maven repository](https://mvnrepository.com/):  
+We will publish it into the maven central repository, which names like `gravitino-filesystem-hadoop3-runtime-{version}-SNAPSHOT.jar`.
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:76]: Why do we use SNAPSHOT jar here?

Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:76]: Could you avoid using we?

File: docs/how-to-use-gvfs.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+```shell
+--conf spark.hadoop.fs.AbstractFileSystem.gvfs.impl=com.datastrato.gravitino.filesystem.hadoop3.Gvfs
+--conf spark.hadoop.fs.gvfs.impl=com.datastrato.gravitino.filesystem.hadoop3.GravitinoVirtualFileSystem
+--conf spark.hadoop.fs.gravitino.server.uri=${your_gravitino_server_uri}
+--conf spark.hadoop.fs.gravitino.client.metalake=${your_gravitino_metalake}
+```
+
+#### 3.3 Operate the fileset storage in your code
+Finally, you can operate the fileset storage in your Spark program:
+```text
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:135]: `text` -> `scala`

**Comment by xloya** [comment]
> Could you polish the document according to #2781?

Yeah, I am only open the PR and will make some modifications based on this document.

File: docs/how-to-use-gvfs.md (hunk 1 above)
Comment (xloya) [reply docs/how-to-use-gvfs.md:34]: Currently will use this config to initialize the Gravitino Client:
```
GravitinoClient.builder(serverUri).withMetalake(metalakeName).withSimpleAuth().build();
```

File: docs/how-to-use-gvfs.md [hunk 5]
```diff
@@ -73,7 +73,7 @@ FileSystem fs = filesetPath.getFileSystem(conf);
 
 Please get the `Gravitino Virtual File System runtime jar` firstly, you can get it in two ways:
 1. [Maven repository](https://mvnrepository.com/):  
-We will publish it into the maven central repository, which names like `gravitino-filesystem-hadoop3-runtime-{version}-SNAPSHOT.jar`.
+It will be published into the maven central repository, which names like `gravitino-filesystem-hadoop3-runtime-{version}.jar`.
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:76]: We should avoid using passive sentences, too.

File: docs/how-to-use-gvfs.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
+---
+title: How to use Gravitino Virtual File System
+slug: /how-to-use-gvfs
+license: "Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2."
+---
+
+## Introduction
+
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:9]: The Gravitino Virtual File System (Gvfs) is a distributed file system mounted into a Gravitino metalake. Gvfs is an abstraction on top of scalable object storage that maps Unix-like filesystem calls to native cloud storage API calls.

File: docs/how-to-use-gvfs.md (hunk 1 above)
Comment (qqqttt123) [reply docs/how-to-use-gvfs.md:34]: > I'm not sure `fs.gravitino.client.metalake` is a config option name.

Sorry, I mean that `fs.gravitino.client.metalake` is a good config option.

**Comment by jerryshao** [comment]
Hi @xloya thanks a lot for your work. I can help you to add more content to this doc.

File: docs/how-to-use-gvfs.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+Gravitino supports operating the storage mapping to `fileset` through the Hadoop Compatible File System (HCFS) interface,
+which names `Gravitino Virtual File System` (It is called `gvfs` for short).  
+You can access the storage mapping to fileset through the following path:
+```text
+gvfs://fileset/${catalog_name}/${schema_name}/${fileset_name}/sub_dir/
+```
+The following mainly introduces how to use `Gravitino Virtual File System` to operate fileset which mapping to HDFS.
+
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:17]: What can you do with Gvfs?

File: docs/how-to-use-gvfs.md [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:18]: What is the Gvfs root?

File: docs/how-to-use-gvfs.md [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+## Prerequisites
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:19]: What about safety of Gvfs?

**Comment by xloya** [comment]
> Hi @xloya thanks a lot for your work. I can help you to add more content to this doc.

Sure, please see what needs to be added.

File: docs/how-to-use-gvfs.md (hunk 1 above)
Comment (qqqttt123) [reply docs/how-to-use-gvfs.md:34]: Think twice. It may be ok. I mark this comment resolved.

**Comment by jerryshao** [comment]
@xloya @coolderli @qqqttt123 I slightly update the doc, would you please help to review?

**Comment by jerryshao** [comment]
I think we should also add Tensorflow support in this doc, WDYT @coolderli ?

**Comment by coolderli** [comment]
> I think we should also add Tensorflow support in this doc, WDYT @coolderli ?

@jerryshao Yes. I think I can add a commit about tensorflow. And you can improve it.

File: docs/how-to-use-gvfs.md [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+          .appName("Gvfs Example")
+          .getOrCreate()
+
+    val rdd = spark.sparkContext.textFile("gvfs://fileset/test_catalog/test_schema/test_fileset_1")
+
+    rdd.foreach(println)
+    ```
+
+
+### Use GVFS with Tensorflow.
```
Comment (coolderli) [inline docs/how-to-use-gvfs.md:188]: @jerryshao I have added docs about TensorFlow. Please take a look and improve it if needed.

File: docs/how-to-use-gvfs.md [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+        <value>test_metalake</value>
+      </property>
+    ```
+
+## How to use
+
+Please make sure to have the Gravitino Virtual File System runtime jar firstly, you can get it in
+two ways:
+
+1. [Maven repository](https://mvnrepository.com/): to download the runtime jar in the Maven central
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:97]: Download from the maven central repo: You can get the package which name is like `gravitino-filesystem-hadoop3-runtime-{version}.jar` from the [Maven repository](https://mvnrepository.com/).

File: docs/how-to-use-gvfs.md [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+   repository, which names like `gravitino-filesystem-hadoop3-runtime-{version}.jar`.
+2. Compile from the source code:
+
+   Download the [Gravitino source code](https://github.com/datastrato/gravitino), and compile it
+   locally using the following command:
+
+    ```shell
+       ./${GRAVITINO_SOURCE_CODE_HOME}/gradlew :clients:filesystem-hadoop3-runtime:build -x test
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:105]: Maybe we should use ./gradlew compileDistribution command

**Comment by qqqttt123** [comment]
Others are ok for me. Thanks for your contribution.

File: docs/how-to-use-gvfs.md [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+kinit -kt your_kerbers.keytab your_kerberos@xxx.com
+
+# 4. Try to list the fileset
+./${HADOOP_HOME}/bin/hadoop dfs -ls gvfs://fileset/test_catalog/test_schema/test_fileset_1
+```
+
+### Use GVFS via Java code
+
+You can also operate the files or directories managed by fileset through Java code.
+Please be sure that your code is running in the correct Hadoop environment, and the environment
```
Comment (xloya) [inline docs/how-to-use-gvfs.md:131]: be sure -> ensure

File: docs/how-to-use-gvfs.md [hunk 14]
```diff
@@ -94,15 +94,15 @@ You can configure these properties in two ways:
 Please make sure to have the Gravitino Virtual File System runtime jar firstly, you can get it in
 two ways:
 
-1. [Maven repository](https://mvnrepository.com/): to download the runtime jar in the Maven central
-   repository, which names like `gravitino-filesystem-hadoop3-runtime-{version}.jar`.
+1. Download from the maven central repository: You can download the runtime jar which names like
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:97]: which name is like.

File: docs/how-to-use-gvfs.md [hunk 15]
```diff
@@ -94,15 +94,15 @@ You can configure these properties in two ways:
...
+   `gravitino-filesystem-hadoop3-runtime-{version}.jar` from [Maven repository](https://mvnrepository.com/).
 2. Compile from the source code:
 
    Download the [Gravitino source code](https://github.com/datastrato/gravitino), and compile it
-   locally using the following command:
+   locally using the following command in the Gravitino source code directory:
 
     ```shell
-       ./${GRAVITINO_SOURCE_CODE_HOME}/gradlew :clients:filesystem-hadoop3-runtime:build -x test
+       ./gradlew :clients:filesystem-hadoop3-runtime:build -x test
```
Comment (qqqttt123) [inline docs/how-to-use-gvfs.md:105]: `compileDistribution` supports the filesystem jar?

Comment (jerryshao) [reply docs/how-to-use-gvfs.md:105]: I think compileDistribution only support packaging the Gravitino server, this is a client-side library, so it is unrelated.

**Comment by jerryshao** [comment]
I've renamed the package from hadoop3 to hadoop, WDYT? @xloya @coolderli

File: docs/how-to-use-gvfs.md (hunk 15 above)
Comment (qqqttt123) [reply docs/how-to-use-gvfs.md:105]: I mean that  
```
./gradlew assembleTrinoConnector
```
I have dedicated command to compile trino connector. We should have similar command for file system in my opinion.

**Comment by qqqttt123** [comment]
> I've renamed the package from hadoop3 to hadoop, WDYT? @xloya @coolderli

+1 for me. Hadoop is better than hadoop3.

File: docs/how-to-use-gvfs.md (hunk 15 above)
Comment (jerryshao) [reply docs/how-to-use-gvfs.md:105]: It's not necessary, we already have a runtime uber jar which can be used.

Comment (qqqttt123) [reply docs/how-to-use-gvfs.md:105]: OK.

**Comment by qqqttt123** [comment]
LGTM.

**Comment by xloya** [comment]
> I've renamed the package from hadoop3 to hadoop, WDYT? @xloya @coolderli

It's okay.

**Comment by coolderli** [comment]
> > I've renamed the package from hadoop3 to hadoop, WDYT? @xloya @coolderli
> 
> +1 for me. Hadoop is better than hadoop3.

+1 for me.

**Pull request merged**